### PR TITLE
OADP-1151: Skip Deletion of ReplicationSource CRs

### PR DESCRIPTION
**Pull request by shubham-pampattiwar**
Images you can use for testing:
```
dataMoverImageFqin: 'quay.io/spampatt/volume-snapshot-mover:keep-rs'
vsmPluginImageFqin: 'quay.io/spampatt/velero-plugin-for-vsm:dia-rs'
```

Related plugin PR: https://github.com/migtools/velero-plugin-for-vsm/pull/16

### CODE REVIEW

File: controllers/cleanup.go [hunk 1]
```diff
@@ -20,7 +20,6 @@ var cleanupVSBTypes = []client.Object{
 	&snapv1.VolumeSnapshot{},
 	&snapv1.VolumeSnapshotContent{},
 	&corev1.Secret{},
-	&volsyncv1alpha1.ReplicationSource{},
```
Comment (weshayutin) [inline controllers/cleanup.go:23]: ah man.. I could done that :) /me runs

**Review by weshayutin — COMMENTED** [review]
Thanks .. afaik looks good

File: controllers/cleanup.go (hunk 1 above)
Comment (shubham-pampattiwar) [reply controllers/cleanup.go:23]: Bingo

**Comment by weshayutin** [comment]
[~spampatt@redhat.com] 

After evoking the manual trigger by editing

triger.manual

In the expiry time frame the snapcontent was removed.
{code:java}
    storageClassName: gp2-csi
    volumeSnapshotClassName: csi-aws-vsc-test
  sourcePVC: snapcontent-259d65fb-eaca-47c9-a139-207fa46622d1-pvc
  trigger:
    manual: vsb-vnzxo
status:
  conditions:
  - lastTransitionTime: "2023-05-16T21:55:23Z"
    message: PersistentVolumeClaim "snapcontent-259d65fb-eaca-47c9-a139-207fa46622d1-pvc"
      not found
    reason: Error
    status: "False"
    type: Synchronizing
  lastManualSync: vsb-vnzxr-trigger
  lastSyncDuration: 23.569855118s
  lastSyncStartTime: "2023-05-16T21:55:23Z"
  lastSyncTime: "2023-05-16T21:40:26Z"
  latestMoverStatus:
    logs: |-
      no parent snapshot found, will read all files
      Added to the repository: 1.674 KiB (756 B stored)
      processed 3 files, 11 B in 0:00
      snapshot fafae839 saved
      Restic completed in 8s
    result: Successful
  restic: {}
 {code}
 

 
{code:java}
**** S3 ****
                          DIR  s3://cvpbucket/openshift-adp/snapcontent-0e3ed10f-90cb-4a64-abb1-151b112f57b6-pvc/
                          DIR  s3://cvpbucket/openshift-adp/snapcontent-21dade7d-ffd3-444a-ae4e-742b25e7a505-pvc/
                          DIR  s3://cvpbucket/openshift-adp/snapcontent-259d65fb-eaca-47c9-a139-207fa46622d1-pvc/
                          DIR  s3://cvpbucket/openshift-adp/snapcontent-2a868598-0bf1-405b-90d9-d6cf4b31dcc2-pvc/
                          DIR  s3://cvpbucket/openshift-adp/snapcontent-3b572fad-baf0-4cc4-9e2f-b031bf9d0383-pvc/
                          DIR  s3://cvpbucket/openshift-adp/snapcontent-40c15450-15a0-4244-b228-be80be8a5b19-pvc/
                          DIR  s3://cvpbucket/openshift-adp/snapcontent-4b7474de-8df1-4dda-9d52-f264a34b750b-pvc/
                          DIR  s3://cvpbucket/openshift-adp/snapcontent-9ff51252-7de1-4a57-9594-f192ec8aec65-pvc/
                          DIR  s3://cvpbucket/openshift-adp/snapcontent-f425de23-b988-4452-98e0-290e4cfe7a2e-pvc/
^C
[whayutin@thinkdoe minimal-8csivol]$ while true; do echo -e "RESTORE ONE \n;" echo "***** VSB *****"; oc get vsb -A; echo "***** VSC *****"; oc get vsc -A; echo "***** VS *****";  oc get vs -A; echo "***** VSR *****"; oc get vsr -A;  echo "**** S3 ****"; s3cmd ls s3://cvpbucket/openshift-adp/; sleep 15; done^C
[whayutin@thinkdoe minimal-8csivol]$ ls s3://cvpbucket/openshift-adp/
ls: cannot access 's3://cvpbucket/openshift-adp/': No such file or directory
[whayutin@thinkdoe minimal-8csivol]$ ls s3://cvpbucket/
ls: cannot access 's3://cvpbucket/': No such file or directory
[whayutin@thinkdoe minimal-8csivol]$ s3cmd ls s3://cvpbucket/openshift-adp/
                          DIR  s3://cvpbucket/openshift-adp/snapcontent-0e3ed10f-90cb-4a64-abb1-151b112f57b6-pvc/
                          DIR  s3://cvpbucket/openshift-adp/snapcontent-21dade7d-ffd3-444a-ae4e-742b25e7a505-pvc/
                          DIR  s3://cvpbucket/openshift-adp/snapcontent-259d65fb-eaca-47c9-a139-207fa46622d1-pvc/
                          DIR  s3://cvpbucket/openshift-adp/snapcontent-2a868598-0bf1-405b-90d9-d6cf4b31dcc2-pvc/
                          DIR  s3://cvpbucket/openshift-adp/snapcontent-3b572fad-baf0-4cc4-9e2f-b031bf9d0383-pvc/
                          DIR  s3://cvpbucket/openshift-adp/snapcontent-40c15450-15a0-4244-b228-be80be8a5b19-pvc/
                          DIR  s3://cvpbucket/openshift-adp/snapcontent-4b7474de-8df1-4dda-9d52-f264a34b750b-pvc/
                          DIR  s3://cvpbucket/openshift-adp/snapcontent-9ff51252-7de1-4a57-9594-f192ec8aec65-pvc/
                          DIR  s3://cvpbucket/openshift-adp/snapcontent-f425de23-b988-4452-98e0-290e4cfe7a2e-pvc/
[whayutin@thinkdoe minimal-8csivol]$ 
 {code}

**Review by weshayutin — APPROVED** [review]

**Comment by shubham-pampattiwar** [comment]
Thanks for the update @weshayutin !

**Review by sseago — APPROVED** [review]

**Review by eemcmullan — APPROVED** [review]

**Pull request merged**